### PR TITLE
NAS-117901 / 22.12 / optimize pools.py alert

### PR DESCRIPTION
**Pull request by yocalebo**
Move the `PoolUpgraded` and `PoolUSBDisks` alerts to be triggered on zfs based events to prevent polling.

### CODE REVIEW

File: src/middlewared/middlewared/alert/source/pools.py [hunk 1]
```diff
@@ -9,22 +9,41 @@ class PoolUSBDisksAlertClass(AlertClass):
...
     schedule = CrontabSchedule(hour=0)  # every 24 hours
     run_on_backup_node = False
 
     async def check(self):
         alerts = []
+        usb_disks = {}
+        not_upgraded_zpools = []
+
+        for guid, info in (await self.middleware.call('zfs.pool.query_imported_fast')).items():
+            if info['state'] != 'ONLINE':
```
Comment (william-gr) [inline src/middlewared/middlewared/alert/source/pools.py:33]: Does this mean we wont get this for degraded pools? If so, why

**Review by themylogin — COMMENTED** [review]
Same suggestion... can/should we do these checks on pool import or config change? IIRC we have some mechanism that will group 1000 equal consecutive pool config changes emitted by ZFS into one event so performance won't suffer.

**Comment by yocalebo** [comment]
retest this please

File: src/middlewared/middlewared/alert/source/pools.py [hunk 2]
```diff
@@ -1,30 +1,52 @@
...
+
+    async def is_upgraded(self, pool_name):
+        try:
+            return await self.middleware.call('zfs.pool.is_upgraded', pool_name)
+        except Exception:
+            return False
+
+    async def create(self, args):
+        pool = args['pool_name']
+        if await self.is_upgraded(pool):
```
Comment (themylogin) [inline src/middlewared/middlewared/alert/source/pools.py:48]: Isn't a pool "upgraded" if it has all the feature flags enabled? (and we generally want all pools to be upgraded and we create alerts if they are not upgraded)?

**Review by themylogin — APPROVED** [review]
Nice!

File: src/middlewared/middlewared/alert/source/pools.py (hunk 2 above)
Comment (yocalebo) [reply src/middlewared/middlewared/alert/source/pools.py]: oops! Fixed :)

**Review by themylogin — APPROVED** [review]

**Pull request merged**